### PR TITLE
Add builders for setting optional attributes on HTTP extractors

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**

### CODE REVIEW

File: instrumentation-api/src/test/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpClientAttributesExtractorTest.java [hunk 1]
```diff
@@ -104,10 +104,12 @@ void normal() {
...
-        HttpClientAttributesExtractor.create(
-            new TestHttpClientAttributesGetter(),
-            CapturedHttpHeaders.create(
-                singletonList("Custom-Request-Header"), singletonList("Custom-Response-Header")));
+        HttpClientAttributesExtractor.builder(new TestHttpClientAttributesGetter())
+            .captureHttpHeaders(
+                CapturedHttpHeaders.create(
+                    singletonList("Custom-Request-Header"),
+                    singletonList("Custom-Response-Header")))
+            .build();
```
Comment (trask) [inline instrumentation-api/src/test/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpClientAttributesExtractorTest.java:112]: what do you think of hiding `CapturedHttpHeaders` and exposing `captureHttpRequestHeaders` and `captureHttpResponseHeaders` on the builders?

**Review by trask — APPROVED** [review]

File: instrumentation-api/src/test/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpClientAttributesExtractorTest.java (hunk 1 above)
Comment (mateuszrzeszutek) [reply instrumentation-api/src/test/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpClientAttributesExtractorTest.java:112]: I think I'd probably prefer leaving this as it is, it's a bit simpler this way.

If I added `captureHttpRequestHeaders(List<String>)` I'd have to take the following things into account:
* is it additive? does `. captureHttpRequestHeaders(["abc"]).captureHttpRequestHeaders(["def"])` capture `["abc", "def"]` or does the second call overwrite the first one?
* do the default (configured) values get overwritten, or appended to?

With `captureHttpHeaders(CapturedHttpHeaders)` it's clear (I hope so? 😅 ) that the previous setting is overwritten. (Maybe I should rename it to `setCapturedHttpHeaders` after all...)

Comment (trask) [reply instrumentation-api/src/test/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpClientAttributesExtractorTest.java:112]: cc @anuraaga for any API thoughts

Comment (anuraaga) [reply instrumentation-api/src/test/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpClientAttributesExtractorTest.java:112]: How about

`setCapturedRequestHeaders`
`setCapturedResponseHeaders`

Comment (mateuszrzeszutek) [reply instrumentation-api/src/test/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpClientAttributesExtractorTest.java:112]: I'll do that in a separate PR - this requires changing all our HTTP library instrumentations, so the amount of changes will be pretty significant.

**Pull request merged**